README.md

```diff
@@ -1,3 +1,74 @@
+
+## Introduction ##
+
+`complex` is a rewrite of the [SIMPL](https://www.github.com/bluequartzsoftware/simpl) library upon which [DREAM3D](https://www.github.com/bluequartzsoftware/dream3d) is written. This library aims to be fully C++17 compliant, removes the need for Qt5 at the library level and brings more flexible data organization and grouping options.
+
+## Prerequisites ##
+
+In order to compile `complex` you will need a C++17 compiler suite installed on your computer.
+
++ Compiler
+  + Windows Visual Studio 2017 v141 toolset
+  + macOS 10.15 and Xcode 12.4 or higher
+  + Linux with GCC Version 9.0 or higher
+
+## Install vcpkg ##
+
+The `complex` project uses the [vcpkg](https://www.github.com/microsoft/vcpkg) to manage it's dependent libraries. If this is not already installed on your system then you will need to download and compile it.
+
+### Windows ###
+
+Clone the vcpkg repository into a location that it will be used from. inside your home directory or at `C:/vcpkg` is a reasonable spot. *DO NOT USE A PATH WITH SPACES IN ANY OF THE FOLDERS*.
+
+```(lang-console)
+cd C:/Users/[USERNAME]/Applications
+git clone ssh://git@github.com/microsoft/vcpkg
+```
+
+Use the Windows Explorer to open the vcpkg directory and double click the `bootstrap-vcpkg.bat` file. This will create the `vcpkg.exe` file. You will need to ensure that the path to the `vcpkg.exe` is on your `PATH` environment variable. There are several ways to do this in Windows, select the one that is right for your environment.
+
+### MacOS/Linux ###
+
+Select an install location (the example below uses `/Users/Shared/DREAM3DNX_SDK/`) and then clone, build and install vcpkg.
+
+```(lang-console)
+mkdir -p /Users/Shared/DREAM3DNX_SDK/
+cd /Users/Shared/DREAM3DNX_SDK/
+git clone https://github.com/microsoft/vcpkg
+cd vcpkg
+./bootstrap-vcpkg.sh -disableMetrics
+export PATH=/Users/Shared/DREAM3DNX_SDK/vcpkg:$PATH
+```
+
+## Clone Appropriate Repositories ##
+
+Within the folder DREAM3D-Dev clone both the `complex` and `DREAM3DNX` repositories
+
+Create a location to keep the `complex` repositories and make builds. You can do either in-source our out-of-source builds.
+
+```(lang-console)
+git clone --recursive ssh://git@github.com/bluequartzsoftware/complex
+git clone --recursive ssh://git@github.com:DREAM3D/DREAM3D_Data.git
+```
+
+## Configure complex with CMake (Windows & MSVC) ##
+
+For this example we are going to do an "in-source" build. By default git will ignore some basic names for build directories such as `Debug, Release, x64`. CMake can generate lots of project files from ninja, to nmake to Visual Studio. For this example we are going to use the `ninja` generator so we will need to keep the Debug and Release builds separated.
```

Comment:
This is still an out-of-source build as the build files are not directly next to the source files, and the folder isn't under source control.

Reply:
historically, "in source" means that the build directory is inside the project directory.